tools/dissector/packet-opendds.cpp

```diff
@@ -462,56 +501,141 @@ namespace OpenDDS
+
+        return;
+      }
+      
+      // Set Payload Field/Tree
+      proto_item * payload_item = proto_tree_add_string(
+        ltree, hf_sample_payload, tvb_, offset,
+        (gint) header.message_length_,
+        data_name);
+      proto_tree * contents_tree = proto_item_add_subtree(
+          payload_item, ett_sample_payload);
+
+      // Try to Dissect Payload
       Sample_Dissector *data_dissector =
         Sample_Manager::instance().find (data_name);
-      if (data_dissector == 0)
-        {
-          ACE_DEBUG ((LM_DEBUG,
-                      "DDS_Dissector::dissect_sample_payload: "
-                      "no dissector found for %s \n",
-                      data_name));
-
-          offset += header.message_length_; // skip marshaled data
-        }
-      else
-        {
-          Wireshark_Bundle params = {tvb_, pinfo_, ltree, offset};
-          offset = data_dissector->dissect (params);
+      if (data_dissector == 0) {
+        ACE_DEBUG ((LM_DEBUG,
+                    "DDS_Dissector::dissect_sample_payload: "
+                    "no dissector found for %s \n",
+                    data_name));
+
+        // Mark Packet
+        proto_tree_add_expert_format(
+          ltree,
+          pinfo_,
+          &ei_sample_payload,
+          tvb_,
+          offset,
+          (gint) header.message_length_,
+          "No Dissector Found for %s \n",
+          data_name
+        );
+
+        offset += header.message_length_; // skip marshaled data
+      } else {
+        // Push DCPS Primary Data Type Namespace
+        // Convert, for example, "1::2::3" to "1.2.3"
+        std::string ns;
+        char c = 0xff;
+        bool delimiter = false;
+        for (size_t i = 0; c; i++) {
+          c = data_name[i];
+          if (c == ':') {
+            if (delimiter) {
+              Sample_Base::push_ns(ns);
+              ns = "";
+              delimiter = false;
+            } else {
+              delimiter = true;
+            }
+          } else if (c) { // Any value that's not : or NULL
+            ns.push_back(c);
+            delimiter = false;
+          }
         }
+        Sample_Base::push_ns(ns);
+
+        // Parameters to pass between the recursive dissect calls
+        const guint8 * const_data = tvb_get_ptr(
+          tvb_, offset, header.message_length_
+        );
+        guint8 * data = const_cast<guint8 *>(const_data);
+        size_t size = static_cast<size_t>(ws_tvb_length(tvb_) - offset);
+        Wireshark_Bundle params(
+          (char *) data, size,
+          header.byte_order_ != ACE_CDR_BYTE_ORDER,
+          Serializer::ALIGN_NONE // For now alignment is not supported
+        );
+        params.tvb = tvb_;
+        params.info = pinfo_;
+        params.tree = contents_tree;
+        params.offset = offset;
+
+        /* 
+         * Handle Seg Faults that might be caused by incorrect ITL file
+         *
+         * Check for enviromental variable $OPENDDS_DISSECTORS_SIGSEGV,
```

Comment:
no "al" here, it's an "environment variable"

Reply:
Hmm, you're right. I have always called them that in my head.